app/directives/fileUpload.js

```diff
@@ -0,0 +1,36 @@
+define(['./module'],function(directives) {
+    directives.directive('fileUpload', function() {
+        return {
+            restrict: 'E',
+            controller: ['$scope','$routeParams','FileUploader', function ($scope, $routeParams, FileUploader) {
+                var uploader = $scope.uploader = new FileUploader({
+                    url: 'http://127.0.0.1:8000/api/problems/' + $routeParams.problemID + '/photos',
+                    withCredentials: true,
+                    alias: 'photos'
+                });
+
+                // FILTERS
+
+                uploader.filters.push({
+                    name: 'customFilter',
+                    fn: function (item /*{File|FileLikeObject}*/, options) {
+                        return this.queue.length < 10;
+                    }
+                });
+
+                // CALLBACKS
+
+                uploader.onBeforeUploadItem = function (item) {
+                    item.formData.push({"comments": item.comments});
+                };
+                uploader.onSuccessItem = function (fileItem, response, status, headers) {
+                    //console.info('onSuccessItem', fileItem, response, status, headers);
```

Comment:
if you are leaving this emply eventListeners here, can you please document the exact moment they are triggering?